File: run/bak/cpp/ffi/Test/FibFFICheatcode.cc
#include "purescript.h"
namespace { volatile int dummy; }
FOREIGN_BEGIN( Test_FibFFICheatcode )
exports["runFibFFICheatcode"] = [](const boxed& in) -> boxed { 
    int n = unbox<int>(in); 
    int a = 0, b = 1;
    for (int i = 0; i < n; i++) { int temp = a; a = b; b = temp + b; }
    dummy = a; return a; 
};
FOREIGN_END

File: scratch/records-tast-opportunity-20260916/full-scalar/src/Test/FibFFICheatcode.erl
-module(test_fibFFICheatcode@foreign).
-export([runFibFFICheatcode/1]).
runFibFFICheatcode(N) when N < 2 -> N;
runFibFFICheatcode(N) -> runFibFFICheatcode(N - 1) + runFibFFICheatcode(N - 2).
